v0/destinations/wootric/transform.js

```diff
@@ -0,0 +1,190 @@
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  getIntegrationsObj,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  getFieldValueFromMessage,
+  CustomError
+} = require("../../util");
+const {
+  getAccessToken,
+  retrieveUserId,
+  flattenPayload,
+  formatIdentifyPayload,
+  formatTrackPayload,
+  createUserPayloadBuilder,
+  updateUserPayloadBuilder,
+  createResponsePayloadBuilder,
+  createDeclinePayloadBuilder,
+  checkExistingEmailAndPhone
+} = require("./util");
+
+const responseBuilder = async (payload, endpoint, method, accessToken) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = endpoint;
+    response.headers = {
+      "Content-Type": "application/x-www-form-urlencoded",
+      Authorization: `Bearer ${accessToken}`
+    };
+    response.method = method;
+    response.body.FORM = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new CustomError("Payload could not be constructed", 400);
+};
+
+const identifyResponseBuilder = async (message, destination) => {
+  let payload;
+  let endpoint;
+  let method;
+  let builder;
+
+  const accessToken = await getAccessToken(destination);
+  const userId = getFieldValueFromMessage(message, "userId");
+  const wootricEndUserId = await retrieveUserId(userId, accessToken);
+  // If user already exist we will update it else creates a new user
+  if (!wootricEndUserId) {
+    builder = createUserPayloadBuilder(message);
+    payload = builder.payload;
+    endpoint = builder.endpoint;
+    method = builder.method;
+  } else {
+    builder = updateUserPayloadBuilder(message);
+    payload = builder.payload;
+    endpoint = builder.endpoint.replace("<end_user_id>", wootricEndUserId);
+    method = builder.method;
+  }
+
+  // Throw error if user already exist with given email/phone
+  await checkExistingEmailAndPhone(
+    payload.email,
+    payload.phone_number,
+    userId,
+    accessToken
+  );
+
+  formatIdentifyPayload(payload);
+  flattenPayload(payload, "properties");
```

Comment:
It is a good practice to assign the returned value to some variable and then making changes to the original JSON. By this what I mean is, flatten the payload add to a temporary object, then return it. Here, use the `...` spreading property to merge the flattened payload with old one, and delete the properties key. Reach out to me if any concerns.